custom/icds/management/commands/contact_phone_fix.py

```diff
@@ -0,0 +1,174 @@
+
+        filename = 'to_be_updated_phone_numbers_%s_%s.csv' % (self.db_alias, datetime.utcnow())
+        with open(filename, 'w') as output:
+            cases_iterated = 0
+            writer = csv.writer(output)
+            writer.writerow(CSV_HEADERS)
+            if self.log_progress:
+                self.stdout.write('iterating now')
+            for case in iter_all_rows(reindex_accessor):
+                if self.log_progress and cases_iterated % CASE_ITERATION_COUNT == 0:
+                    self.stdout.write("cases iterated: %s" % cases_iterated)
+                cases_iterated += 1
+                if self._case_needs_to_be_updated(case):
+                    case_ids_with_to_be_updated_phone_number.append(case.case_id)
+                    writer.writerow([case.case_id])
+        return case_ids_with_to_be_updated_phone_number
+
+    def _reassured_case_ids_to_update(self, chunk):
+        # reconfirm the cases before updating to avoid removing updates in between
+        # fetching case ids and updating
+        update_cases = self.case_accessor.get_cases(list(chunk))
+        case_ids_list = []
+        for update_case in update_cases:
+            if (update_case.get_case_property(HAS_MOBILE_PROPERTY) == HAS_MOBILE_PROPERTY_VALUE and 
+                len(update_case.get_case_property(PHONE_NUMBER_PROPERTY)) == 10 and 
+                update_case.get_case_property(CONTACT_PHONE_NUMBER_IS_VERIFIED) == CONTACT_PHONE_NUMBER_IS_VERIFIED_VALUE):
+                case_ids_list.append(update_case.case_id)
+        return case_ids_list
+
+    def _submit_update_form(self, case_ids_list, exceptions_raised):
+        update_case_blocks = create_case_blocks(case_ids_list)
+        for attempt in range(MAX_RESCUE_EXCEPTIONS_ON_UPDATE):
+            try:
+                if update_case_blocks:
+                    submit_case_blocks(update_case_blocks, DOMAIN, user_id=SYSTEM_USER_ID)
+            except Exception as e:
+                exc = sys.exc_info()
+                exceptions_raised += 1
+                if self.log_progress:
+                    self.stdout.write("rescuing exception %s %s" % (exceptions_raised, str(e)))
+                if exceptions_raised > MAX_RESCUE_EXCEPTIONS_ON_UPDATE:
+                    six.reraise(*exc)
+                else:
+                    time.sleep(60)  # wait for 1 min before trying again
+            else:
+                break
+        return exceptions_raised
+
+    def _update_cases(self, case_ids_with_to_be_updated_phone_number):
+        exceptions_raised = 0
+        with open('to_be_updated_phone_numbers_%s_updated.csv' % self.db_alias, 'w') as output:
+            writer = csv.writer(output)
+            writer.writerow(['Case Id'])
+            case_ids_to_update_chunk = list(chunked(case_ids_with_to_be_updated_phone_number, 100))
+            for chunk in with_progress_bar(case_ids_to_update_chunk):
+                case_ids_list = self._reassured_case_ids_to_update(chunk)
```

Comment:
the chunk here are ids and not case objects. You would need to fetch those cases again first by using accessor.get_cases and use that for both `_reassured_case_ids_to_update` and `_submit_update_form` to work